pkg/cmd/update/update.go

```diff
@@ -0,0 +1,286 @@
+package update
+
+import (
+	"archive/tar"
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/charmbracelet/glamour"
+	"github.com/coreeng/corectl/pkg/cmdutil/config"
+	"github.com/coreeng/corectl/pkg/cmdutil/shell"
+	"github.com/coreeng/corectl/pkg/cmdutil/userio"
+	"github.com/coreeng/corectl/pkg/cmdutil/userio/confirmation"
+	"github.com/coreeng/corectl/pkg/git"
+	"github.com/coreeng/corectl/pkg/version"
+	"github.com/google/go-github/v59/github"
+	"github.com/phuslu/log"
+	"github.com/spf13/cobra"
+)
+
+type UpdateOpts struct {
+	githubToken      string
+	streams          userio.IOStreams
+	targetVersion    string
+	skipConfirmation bool
+}
+
+// Any failures we recieve will log a warning, we don't want this to cause any command to fail, this is an optional
+// check which shouldn't prevent or interrupt any command from running (especially in ci)
+func CheckForUpdates(cfg *config.Config, cmd *cobra.Command) {
+	updateInterval := 1 * time.Hour
+	updateStatusFileName := "corectl-autoupdate"
+	log.Debug().Msg("checking for updates")
+	nonInteractive, err := cmd.Flags().GetBool("non-interactive")
+	if err != nil {
+		log.Warn().Err(err).Msg("could not get non-interactive flag")
+		return
+	}
+
+	if nonInteractive {
+		log.Debug().Msg("skipping update check for --non-interactive command")
+		return
+	}
+
+	tempDir := os.TempDir()
+	tempFilePath := filepath.Join(tempDir, updateStatusFileName)
+	file, err := os.OpenFile(tempFilePath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		log.Warn().Err(err).Msgf("could not open file to set update status %s", tempFilePath)
+		return
+	}
+	defer file.Close()
+
+	data, err := os.ReadFile(tempFilePath)
+	if err != nil {
+		log.Warn().Err(err).Msgf("could not read timestamp from update status file: %s", tempFilePath)
+		return
+	}
+
+	now := time.Now()
+	previousTimeString := string(data)
+	previousTime, err := time.Parse(time.RFC3339, previousTimeString)
+	// This is expected to fail on first run
+	if err != nil {
+		log.Debug().Err(err).Msgf("could not parse timestamp from update status file: %s previousTimeString: \"%s\"", tempFilePath, previousTimeString)
+		// go's time.Sub only works with time.Time, not time.Duration
+		previousTime = now.Add(-updateInterval)
+	}
+
+	timeSince := now.Sub(previousTime)
+	if timeSince >= updateInterval {
+		githubToken := ""
+		if cfg.IsPersisted() {
```

Comment:
I think you can just directly get the value from the config, without this check. In the worst case it will be an empty string. Otherwise it's a value from the config file or from the passed flag (can it be the case?)